src/suricata-plugin.h

```diff
@@ -0,0 +1,67 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#ifndef __SURICATA_PLUGIN_H__
+#define __SURICATA_PLUGIN_H__
+
+#include "config.h"
+
+#ifdef SURICATA_PLUGIN
```

Comment:
why is this needed?

Reply:
I think this should be temporary until the includes that are required by the plugins can be cleanly separated from the includes required by Suricata proper. This specific instance was to save plugins from needing to specify where to have the Lua and NSS headers if they were not picked up by default -- something a plugin really shouldn't have to worry about. However, I think I resolved this in this version of the PR so this instance of this can be removed.